src/gt4py/storage/utils.py

```diff
@@ -217,44 +215,70 @@ def allocate_gpu_unmanaged(default_origin, shape, layout_map, dtype, alignment_b
     if device_field.ndim > 0:
         device_field = device_field[tuple(slice(0, s, None) for s in shape)]
 
-    return raw_buffer, field, device_raw_buffer, device_field
+    return device_raw_buffer, device_field
 
 
-def allocate_cpu(default_origin, shape, layout_map, dtype, alignment_bytes):
-    def allocate_f(size, dtype):
-        raw_buffer = np.empty(size, dtype)
-        return raw_buffer, raw_buffer
+def allocate_cpu(aligned_index, shape, layout_map, dtype, alignment_bytes):
+    dtype = np.dtype(dtype)
+    assert (
+        alignment_bytes % dtype.itemsize
+    ) == 0, "Alignment must be a multiple of byte-width of dtype."
```

Comment:
Unless this check uses `assert` on purpose for performance reasons, I think it'd be better to use `if` and raise an exception here to avoid it can be disabled in performance mode.